basicswap/interface/firo.py

```diff
@@ -102,6 +102,70 @@ def isAddressMine(self, address: str, or_watch_only: bool = False) -> bool:
             return addr_info["ismine"]
         return addr_info["ismine"] or addr_info["iswatchonly"]
 
+    def getNewSparkAddress(self) -> str:
+        """Generate a new Spark address for receiving private funds.
+        RPC: getnewsparkaddress
+        """
+        try:
+            return self.rpc_wallet("getnewsparkaddress")[0]
+        except Exception as e:
+            self._log.error(f"getnewsparkaddress failed: {str(e)}")
+            raise
+
+    def getNewStealthAddress(self):
+        """Get a new Spark address (alias for consistency with other coins)."""
+        return self.getNewSparkAddress()
+
+    def getWalletInfo(self):
+        """Get wallet info including Spark balance."""
+        rv = super(FIROInterface, self).getWalletInfo()
+        try:
+            spark_balance_info = self.rpc("getsparkbalance")
+            # getsparkbalance returns a dict with confirmed, unconfirmed, immature
+            # Values are in FIRO (not satoshis), similar to getwalletinfo balance
+            rv["spark_balance"] = spark_balance_info.get("confirmed", 0)
+            rv["spark_unconfirmed"] = spark_balance_info.get("unconfirmed", 0)
+            rv["spark_immature"] = spark_balance_info.get("immature", 0)
```

Comment:
```suggestion
            rv["spark_balance"] = spark_balance_info.get("confirmedBalance", 0)
            rv["spark_unconfirmed"] = spark_balance_info.get("unconfirmedBalance", 0)
            rv["spark_immature"] = spark_balance_info.get("fullBalance", 0)
```

Amounts are displayed, but are in atomic units (so 100000000 is displayed when you have 1 firo)